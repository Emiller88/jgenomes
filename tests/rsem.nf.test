nextflow_pipeline {

    name "Build rsem references"
    script "../main.nf"
    tag "pipeline"

    test("-profile test [rsem] --input assets/test/default_extended.yml") {

        options "-output-dir $outputDir"

        when {
            params {
                input  = "${projectDir}/assets/test/default_extended.yml"
                outdir = "$outputDir"
                tools  = 'rsem,rsem_make_transcript_fasta'
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_references_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }

    test("-profile test [rsem] --input GRCh38_chr21.yml") {

        options "-output-dir $outputDir"

        when {
            params {
                input  = 'https://raw.githubusercontent.com/nf-core/test-datasets/references/references/GRCh38_chr21/GRCh38_chr21.yml'
                outdir = "$outputDir"
                tools  = 'rsem,rsem_make_transcript_fasta'
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_references_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }
}
