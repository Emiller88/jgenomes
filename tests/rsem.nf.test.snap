{
    "Run with profile test for rsem | --input assets/genomes/test/pipelines/default_extended.yml": {
        "content": [
            2,
            {
                "RSEM_PREPAREREFERENCE_GENOME": {
                    "rsem": "1.3.1",
                    "star": "2.7.10a"
                },
                "Workflow": {
                    "nf-core/references": "v1.0dev"
                }
            },
            [
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_software_versions.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf_core_references_software_mqc_versions.yml",
                "rsem",
                "rsem/rsem",
                "rsem/rsem/GRCh38_chr21.fa",
                "rsem/rsem/Genome",
                "rsem/rsem/Log.out",
                "rsem/rsem/SA",
                "rsem/rsem/SAindex",
                "rsem/rsem/chrLength.txt",
                "rsem/rsem/chrName.txt",
                "rsem/rsem/chrNameLength.txt",
                "rsem/rsem/chrStart.txt",
                "rsem/rsem/exonGeTrInfo.tab",
                "rsem/rsem/exonInfo.tab",
                "rsem/rsem/geneInfo.tab",
                "rsem/rsem/genome.chrlist",
                "rsem/rsem/genome.grp",
                "rsem/rsem/genome.idx.fa",
                "rsem/rsem/genome.n2g.idx.fa",
                "rsem/rsem/genome.seq",
                "rsem/rsem/genome.ti",
                "rsem/rsem/genome.transcripts.fa",
                "rsem/rsem/genomeParameters.txt",
                "rsem/rsem/sjdbInfo.txt",
                "rsem/rsem/sjdbList.fromGTF.out.tab",
                "rsem/rsem/sjdbList.out.tab",
                "rsem/rsem/transcriptInfo.tab"
            ],
            [
                "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f",
                "GRCh38_chr21.fa:md5,c675070fcf168d7b64cfadc30b5d7b4d",
                "Genome:md5,612664e3cfde5e1b73ad541d93752b31",
                "SA:md5,074ae54177bb7b9cb981382f043f36e5",
                "SAindex:md5,81a04dbb4a8366c02907ad35218d877e",
                "chrLength.txt:md5,b0be0a56ddefa84552742c72d4859eac",
                "chrName.txt:md5,e99d7d1051eee43ceab5563c2d09fcee",
                "chrNameLength.txt:md5,c985a141685e8431ec27c782816cb744",
                "chrStart.txt:md5,6925b594ea2eeb964ba87cd6d42ab98f",
                "exonGeTrInfo.tab:md5,ea42dd46e177f95a72a8f6a2925f7348",
                "exonInfo.tab:md5,42a3ce2024f396b2298c0a84b07fb6e4",
                "geneInfo.tab:md5,d0092fb1ea6fcdac270c4df788de70fb",
                "genome.chrlist:md5,c985a141685e8431ec27c782816cb744",
                "genome.grp:md5,f75057a7bf6943f88c04f164cd641105",
                "genome.idx.fa:md5,fd4e06fc250a19351a611a44861b9aea",
                "genome.n2g.idx.fa:md5,fd4e06fc250a19351a611a44861b9aea",
                "genome.seq:md5,03f87c5a63ec7cb6e681e8e4e042153e",
                "genome.ti:md5,38ba06dd31c6feaa10c8d9b42662358c",
                "genome.transcripts.fa:md5,fd4e06fc250a19351a611a44861b9aea",
                "genomeParameters.txt:md5,18fd87d37983d4d39bef7e9b773b5c23",
                "sjdbInfo.txt:md5,12fb05dc7cea89735a0c19e1c0df61cb",
                "sjdbList.fromGTF.out.tab:md5,5d9761b49920fb70a77d74e390d196b9",
                "sjdbList.out.tab:md5,766fbca932681f8666b3a9e5fb3640bd",
                "transcriptInfo.tab:md5,21c1f470ffe3b55b23900e7b7eaec2f4"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-27T11:05:05.450221738"
    },
    "Run with profile test for rsem | --input GRCh38_chr21.yml": {
        "content": [
            4,
            {
                "GFFREAD": {
                    "gffread": "0.12.7"
                },
                "MAKE_TRANSCRIPTS_FASTA": {
                    "rsem": "1.3.1",
                    "star": "2.7.10a"
                },
                "RSEM_PREPAREREFERENCE_GENOME": {
                    "rsem": "1.3.1",
                    "star": "2.7.10a"
                },
                "Workflow": {
                    "nf-core/references": "v1.0dev"
                }
            },
            [
                "gffread",
                "gffread/GRCh38_chr21.gtf",
                "make",
                "make/genome.transcripts.fa",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_software_versions.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf_core_references_software_mqc_versions.yml",
                "rsem",
                "rsem/rsem",
                "rsem/rsem/GRCh38_chr21.fa",
                "rsem/rsem/Genome",
                "rsem/rsem/Log.out",
                "rsem/rsem/SA",
                "rsem/rsem/SAindex",
                "rsem/rsem/chrLength.txt",
                "rsem/rsem/chrName.txt",
                "rsem/rsem/chrNameLength.txt",
                "rsem/rsem/chrStart.txt",
                "rsem/rsem/exonGeTrInfo.tab",
                "rsem/rsem/exonInfo.tab",
                "rsem/rsem/geneInfo.tab",
                "rsem/rsem/genome.chrlist",
                "rsem/rsem/genome.grp",
                "rsem/rsem/genome.idx.fa",
                "rsem/rsem/genome.n2g.idx.fa",
                "rsem/rsem/genome.seq",
                "rsem/rsem/genome.ti",
                "rsem/rsem/genome.transcripts.fa",
                "rsem/rsem/genomeParameters.txt",
                "rsem/rsem/sjdbInfo.txt",
                "rsem/rsem/sjdbList.fromGTF.out.tab",
                "rsem/rsem/sjdbList.out.tab",
                "rsem/rsem/transcriptInfo.tab"
            ],
            [
                "GRCh38_chr21.gtf:md5,c552c0aee5f1becb1d3991d21e63a5c1",
                "genome.transcripts.fa:md5,fd4e06fc250a19351a611a44861b9aea",
                "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f",
                "GRCh38_chr21.fa:md5,c675070fcf168d7b64cfadc30b5d7b4d",
                "Genome:md5,612664e3cfde5e1b73ad541d93752b31",
                "SA:md5,074ae54177bb7b9cb981382f043f36e5",
                "SAindex:md5,81a04dbb4a8366c02907ad35218d877e",
                "chrLength.txt:md5,b0be0a56ddefa84552742c72d4859eac",
                "chrName.txt:md5,e99d7d1051eee43ceab5563c2d09fcee",
                "chrNameLength.txt:md5,c985a141685e8431ec27c782816cb744",
                "chrStart.txt:md5,6925b594ea2eeb964ba87cd6d42ab98f",
                "exonGeTrInfo.tab:md5,a36b92eeceaaf921b9d19cd8e806a98e",
                "exonInfo.tab:md5,bb892190d1ebdd59bf55652916121479",
                "geneInfo.tab:md5,4303e97d841035b5c0fcc3686b2f5a36",
                "genome.chrlist:md5,c985a141685e8431ec27c782816cb744",
                "genome.grp:md5,f75057a7bf6943f88c04f164cd641105",
                "genome.idx.fa:md5,fd4e06fc250a19351a611a44861b9aea",
                "genome.n2g.idx.fa:md5,fd4e06fc250a19351a611a44861b9aea",
                "genome.seq:md5,03f87c5a63ec7cb6e681e8e4e042153e",
                "genome.ti:md5,283306157cf36e6827e4274c896844ba",
                "genome.transcripts.fa:md5,fd4e06fc250a19351a611a44861b9aea",
                "genomeParameters.txt:md5,f26eb5881ea7e6fe1ad0c58e635a5a3d",
                "sjdbInfo.txt:md5,12fb05dc7cea89735a0c19e1c0df61cb",
                "sjdbList.fromGTF.out.tab:md5,d9e4a184cde15a5ab282ff66e740a4a2",
                "sjdbList.out.tab:md5,766fbca932681f8666b3a9e5fb3640bd",
                "transcriptInfo.tab:md5,ad9baa68c5432908b42693edb9aed02a"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-11-28T10:01:45.020113604"
    }
}